packages/core/src/modules/vc/data-integrity/W3cJsonLdCredentialService.ts

```diff
@@ -265,6 +304,52 @@ export class W3cJsonLdCredentialService {
         challenge: options.challenge,
         domain: options.domain,
         documentLoader: this.w3cCredentialsModuleConfig.documentLoader(agentContext),
+        checkStatus: async ({ credential }: { credential: W3cJsonCredential }) => {
```

Comment:
Can you extract this into a method and reuse that for both checkStatus methods?

Reply:
@TimoGlastra I have developed a common function for this purpose. However, in the **checkStatus** method, there are specific conditions related to credential status that are dependent on the payload received from the **verifyCredential** and **verifyPresentation** functions. Therefore, the **checkStatus** method invokes the common function, as it contains the identical logic.